src/ghga_transpiler/core.py

```diff
@@ -33,20 +51,28 @@ class GHGAWorkbook:
     def __init__(self, workbook: Workbook, configs_package: resources.Package):
         """Create a new GHGAWorkbook object from an XLSX workbook"""
         self.workbook = workbook
-        self.version = GHGAWorkbook._get_version(workbook)
-        self.config = config.load_config(self.version, configs_package)
+        self.wb_version = GHGAWorkbook._get_version(workbook)
+        self.config = config.load_config(self.major_minor_version, configs_package)
 
     @staticmethod
     def _get_version(workbook):
         """Function to get workbook version from the worksheet _properties"""
         if "__properties" in workbook.sheetnames:
-            version = str(workbook["__properties"].cell(1, 1).value)
-            if re.fullmatch(SEMVER_REGEX, version):
-                return version
+            try:
+                return SemanticVersion(workbook["__properties"].cell(1, 1).value)
+            except InvalidVersion:
+                raise InvalidSematicVersion(
+                    "Not a valid version following semantic versioning"
+                ) from None
         raise SyntaxError(
             "Unable to extract metadata version from the provided workbook."
         )
```

Comment:
The `SyntaxError` is now never reached anymore. Also, the `InvalidSemanticVersion` error isn't caught at the CLI level:
https://github.com/ghga-de/ghga-transpiler/blob/651bba8d692d59c87b00f9315b5f2090a8a68faf/src/ghga_transpiler/cli.py#L52-L55

Reply:
Do you mean `UnknownVersionError`? Because it falls to the `SyntaxError` when `if "__properties" in workbook.sheetnames` fails.

Reply:
Wouldn't it fall to the `SyntaxError` when `if "__properties" in workbook.sheetnames:` fails? The check is right before the highlighted try, except block.